### PR TITLE
HierarchicalResourceQuota documentation

### PR DESCRIPTION
**Pull request by zfrhv**
solves #264 

i have a feeling my charges are not that great,
@adrianludwin what to you think of it for now?

### CODE REVIEW

**Comment by mochizuki875** [comment]
Hi, @zfrhv
Thanks for your working! 
Could you please add the point to note when using HRQ?(ref #292)
The main point is that decimal point value cannot be specified without `""` unlike `ResourceQuota`.

File: docs/user-guide/quickstart.md [hunk 1]
```diff
@@ -458,6 +459,94 @@ kubectl delete svc s2 -n service-2
 kubectl delete pods s2 -n service-2
 ```
 
+<a name="hrq"/>
+
+### Hierarchical resource quotas
```
Comment (mochizuki875) [inline docs/user-guide/quickstart.md:464]: Hi, @zfrhv
Thanks for your working!
Could you please add the point to note when using HRQ?(ref https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/292)
The main point is that decimal point value cannot be specified without "" unlike ResourceQuota.

File: docs/user-guide/concepts.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ namespaces are, and _why_ they behave the way they do.
 * [Basic concepts](#basic)
   * [Parents, children, trees and forests](#basic-trees)
   * [Full namespaces and subnamespaces](#basic-subns)
+  * [Hierarchical Resource Quotas](#hierarchical-resource-quota)
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:16]: Use lowercase to match other headings, but you can add the acronym. Also, I'd use the acronym as the anchor, e.g.:

`[Hierarchical resource quotas (HRQs)](#hrq)`

File: docs/user-guide/concepts.md [hunk 3]
```diff
@@ -87,6 +88,8 @@ namespaces:
   as isolation units for their own services. However, namespace creation is a
   privileged cluster-level operation, and you typically want to control this
   privilege very closely.
+* You might want to give some amount of resources to a team (similar to ResourceQuota), and they can
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:91]: ResourceQuota in backquotes, e.g. `ResourceQuota`

Also below for `HierarchicalResourceQuota`

File: docs/user-guide/concepts.md [hunk 4]
```diff
@@ -218,6 +221,25 @@ probably not a great idea.
 You can create a subnamespace from the command line via `kubectl hns create child
 -n parent`.
 
+<a name="hierarchical-resource-quota">
+
+### Hierarchical Resource Quotas
+
+When you want to give some amount of resources to `team-a`, and want them to be able
+to distrtibute their resources between themselves, you create a HierarchicalResourceQuota
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:229]: sp: distribute

Maybe instead of "between themselves," you can say "to be able to flexibly use resources in _any_ of their namespaces"

File: docs/user-guide/concepts.md [hunk 5]
```diff
@@ -218,6 +221,25 @@ probably not a great idea.
...
+in namespace `team-a`. any member from `team-a` can create his own HierarchicalResourceQuota
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:230]: "any" -> "Any" (new sentence)

"his own" -> "their own" (not gender-specific)

File: docs/user-guide/concepts.md [hunk 6]
```diff
@@ -218,6 +221,25 @@ probably not a great idea.
...
+in his subnamespace, and the sum of all reasources from all the subnamespaces of the
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:231]: reasources -> resources

File: docs/user-guide/concepts.md [hunk 7]
```diff
@@ -218,6 +221,25 @@ probably not a great idea.
...
+members wont be over the amount of resources that is configured in
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:232]: wont -> will not

File: docs/user-guide/concepts.md [hunk 8]
```diff
@@ -218,6 +221,25 @@ probably not a great idea.
...
+HierarchicalResourceQuota of namespace `team-a`.
+
+when creating HierarchicalResourceQuota, HNC automatically creates ResourceQuota
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:235]: Capitalize: when -> When

But maybe I'd replace this whole paragraph with this:

```
To implement hierarchical quotas, HNC automatically creates `ResourceQuota` objects in each
affected namespace. These objects should not be modified or relied on by users, as they only
track the resources from individual namespaces, not the entire tree. To inspect hierarchical
quotas, look at the HRQ objects, or use the `kubectl hrq` command.
```

File: docs/user-guide/how-to.md [hunk 9]
```diff
@@ -10,6 +10,7 @@ This document describes common tasks you might want to accomplish using HNC.
   * [Create a subnamespace](#use-subns-create)
   * [Inspect namespace hierarchies](#use-inspect)
   * [Propagating policies across namespaces](#use-propagate)
+  * [Limit Resources over parent namespaces](#use-hrq)
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:13]: Suggested: `Apply hierarchical resource quotas (HRQs)` (people will be looking for the word "quota" I think)

File: docs/user-guide/how-to.md [hunk 10]
```diff
@@ -186,6 +187,15 @@ permissions. To understand why an object is not being propagated to a namespace,
 use `kubectl hns describe <ns>`, where `<ns>` is either the source (ancestor) or
 destination (descendant) namespace.
 
+<a name="use-hrq"/>
+
+### Limit Resources over parent namespaces
+
+HNC has an object called HirarchicalResourceQuota that is similar to ResourceQuota
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:194]: "that is similar" -> "which is a drop-in replacement for"

File: docs/user-guide/how-to.md [hunk 11]
```diff
@@ -186,6 +187,15 @@ permissions. To understand why an object is not being propagated to a namespace,
...
+but across all namespace tree. It allows you to distribute tour resources between
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:195]: "across all namespace tree." -> "across all the namespaces in a hierarchy."

"tour resources" -> "your resources"

File: docs/user-guide/how-to.md [hunk 12]
```diff
@@ -186,6 +187,15 @@ permissions. To understand why an object is not being propagated to a namespace,
...
+teams, and those teams can distribute their resources between their subteams...
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:196]: "between their subteams..." -> "flexibly between their namespaces, without worrying about the exact usage in any one namespace."

File: docs/user-guide/quickstart.md [hunk 13]
```diff
@@ -458,6 +459,94 @@ kubectl delete svc s2 -n service-2
...
+    place: holder
+  ports:
+  - protocol: TCP
+    port: 80
+    targetport: 80
+EOF
+```
+We get an error:
+```
+Error from server (Forbidden): error when creating "STDIN": admission webhood "resourcesquotasstatus.hnc.x-k8s.io" denied the request: exceeded hierarchical quota in namespace "acme-org": "acme-org-hrq", requested: services=1, used: services=1, limited: services=1
```
Comment (adrianludwin) [inline docs/user-guide/quickstart.md:533]: Can you add newlines to make this easier to read?

File: docs/user-guide/quickstart.md [hunk 14]
```diff
@@ -458,6 +459,94 @@ kubectl delete svc s2 -n service-2
...
+```
+
+to view the hrq usage simply run
```
Comment (adrianludwin) [inline docs/user-guide/quickstart.md:536]: Capitalize: `To view the HRQ usage, simply run:`

File: docs/user-guide/quickstart.md [hunk 15]
```diff
@@ -458,6 +459,94 @@ kubectl delete svc s2 -n service-2
...
+```bash
+kubectl hns hrq -n acme-org
+```
+you can also biew in all namespace
```
Comment (adrianludwin) [inline docs/user-guide/quickstart.md:540]: Fix capitalization, also "biew" -> "view"

**Review by adrianludwin — COMMENTED** [review]
This is great! Thanks so much for writing this. I've made a bunch of minor copy-edits, can you please make those fixes and force-push them to the existing commit? Then we can get this in. Thanks again!

File: docs/user-guide/quickstart.md (hunk 1 above)
Comment (adrianludwin) [reply docs/user-guide/quickstart.md:464]: sgtm, though please add it either at the _bottom_ of the quickstart and/or in the howto section.

Comment (zfrhv) [reply docs/user-guide/quickstart.md:464]: sure!
sorry idk how i missed that notification
i will add both as it sounds pretty important

File: docs/user-guide/concepts.md (hunk 2 above)
Comment (zfrhv) [reply docs/user-guide/concepts.md:16]: thanks!
you are a pro!

File: docs/user-guide/concepts.md (hunk 4 above)
Comment (zfrhv) [reply docs/user-guide/concepts.md:229]: sounds like better english, 
but it also sound like a differente point

i was more trying to tell that they can limit their subnamespaces with hrqs as well (becaues that is exactly why we needed hrqs), 
and not that hrq will work over all of the subnamespaces

for example in this tree:
```
acme-org
├── team-a
│   ├── developer-1
│   └── developer-2
└── team-b
```
`acme-org` will have hrq with 100 cpus, and they can give 60 cpu to `team-a` and 40 cpu to `team-b`,
then `team-a` can give 30 cpu to `developer-1` and other 30 cpus to `developer-2`...
and the whole point is that each section/team can split and distribute their own reasources between themselves without involving the cluster admins.

should i add a chart with example so it would be more clear?

Comment (adrianludwin) [reply docs/user-guide/concepts.md:229]: IMO most of the value in HRQ is that you _don't_ need to do that if you don't want to. For example, without HRQ, we could give the admin of `team-a` the ability to create `ResourceQuotas` in `team-a`, and they could just create a normal `ResourceQuota` in each of `developer-1` and `developer-2`. What HRQ gives you is the ability to share quota _across_ `developer-1` _and_ `developer-2` - that is, just put a single HRQ object in `team-a` and not worry about the details.

Without HRQ, if `team-a` is allowed to use 60 CPUs, _either_ they need to put an RQ of 60 in both `developer-1` and `developer-2`, which means that the team could actually use up to 120 (oops). Or they need put in RQs of 30 in each, as you suggested, which means that even if `developer-1` is only using 1 CPU and `developer-2` needs to use 50, they can't.

In other words, without HRQ, you have to be either too loose or too strict. With HRQ, you you only need to be strict where it really matters (e.g. at the team level) and everything else will just work.

Does that make sense?

File: docs/user-guide/how-to.md (hunk 12 above)
Comment (zfrhv) [reply docs/user-guide/how-to.md:196]: same as i answered in previous note,
it sounds easier to understand, but my point was to not make them think that "hrq can work perfectly across your subnamespaces and you dont need to worry about their sum".
i was trying to explain that each team has their own permissions, and amount of reasources, and they can distribute their own resources as they want between their sub teams without involving anyone else.

File: docs/user-guide/quickstart.md (hunk 13 above)
Comment (zfrhv) [reply docs/user-guide/quickstart.md:533]: sure!
i was struggling should i make the error look as is, or should i make it more clear

File: docs/user-guide/how-to.md (hunk 12 above)
Comment (adrianludwin) [reply docs/user-guide/how-to.md:196]: It might be worth mentioning that if this is what people want, you can use `RoleBindings` to give team admins the permission to create regular `ResourceQuota` objects, which will solve that problem if the goal is to limit consumption in the "leaf" namespaces.

File: docs/user-guide/concepts.md (hunk 4 above)
Comment (zfrhv) [reply docs/user-guide/concepts.md]: > without HRQ, we could give the admin of `team-a` the ability to create ResourceQuotas in `team-a`, and they could just create a normal `ResourceQuota` in each of `developer-1` and `developer-2`.

thats exactly the problem we all struggled with, because if we give to `team-a` the ability to create/manage `ResourceQuotas`, then they have no limit, they can use as much resources as they want.
and the solution is having HRQ in `team-a` namespace that they cannot modify, but they can create HRQs in their subnamespaces (`developer-1` and `developer-2`).
this allows for each team to seperate their resources across their subteams.

i can give more clear example but then it a litle complitaes and has many teams (big tree)

---

> Without HRQ, if team-a is allowed to use 60 CPUs, either they need to put an RQ of 60 in both developer-1 and developer-2, which means that the team could actually use up to 120 (oops). Or they need put in RQs of 30 in each, as you suggested, which means that even if developer-1 is only using 1 CPU and developer-2 needs to use 50, they can't.

thats a good point, but its not what im trying to tell
i think i will try to add both points.
give me 30 min

Comment (adrianludwin) [reply docs/user-guide/concepts.md]: Oh, I see. Hmm. You could still create two RQs in `acme-org` and have one propagate only to `team-a` and the other only to `team-b`, but yes, using HRQs for that is much more elegant.

Comment (zfrhv) [reply docs/user-guide/concepts.md]: now as im thinking about it im not sure if its possible 😕
is it possible to make admin of `team-a` but be able to manage HRQs in the subnamespaces (`developer-1` and `developer-2`) but not in his own namespace?

admin of `team-a` CANT change HRQs in namespace `team-a`
admin of `team-a` CAN change HRQs in subnamespace `developer-1` and `developer-2`

Comment (zfrhv) [reply docs/user-guide/concepts.md]: ok, sorry it took me a while to figure out how its possible to make
what do you think of it now?
(idk how to put comments on the text itself, im pretty new at git)

File: docs/user-guide/concepts.md (hunk 3 above)
Comment (adrianludwin) [reply docs/user-guide/concepts.md:91]: Can you add the backquotes? E.g. instead of:

```
(similar to ResourceQuota)
```

it should look like this:

```
(similar to `ResourceQuota`)
```

And ditto everywhere else in this change where you say either ResourceQuota or HierarchicalResourceQuota.

File: docs/user-guide/concepts.md [hunk 16]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+│   ├── org-a-team-1
+│   ├── org-a-team-2
+│   ...
+├── organization-b
+│   ├── org-b-team-1
+│   ├── org-b-team-2
+│   ...
+...
+```
+Instead of each team askin from the `cluster-admin` to modify their ResourceQuota,
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:250]: askin -> asking

File: docs/user-guide/how-to.md [hunk 17]
```diff
@@ -186,6 +187,17 @@ permissions. To understand why an object is not being propagated to a namespace,
 use `kubectl hns describe <ns>`, where `<ns>` is either the source (ancestor) or
 destination (descendant) namespace.
 
+<a name="use-hrq"/>
+
+### Limit Resources over parent namespaces
+
+HNC has an object called `HirarchicalResourceQuota` that is similar to `ResourceQuota`
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md]: "is similar to" -> "is a drop-in replacement for"

"HirarchicalResourceQuota" -> "HierarchicalResourceQuota" (missing "e" in "Hier")

File: docs/user-guide/how-to.md [hunk 18]
```diff
@@ -186,6 +187,17 @@ permissions. To understand why an object is not being propagated to a namespace,
...
+but across all the namespaces in a hierarchy. It allows you to distribute your resources between
+teams, and those teams can distribute their resources between their subteams...
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:196]: No "..." at the end

File: docs/user-guide/how-to.md [hunk 19]
```diff
@@ -186,6 +187,17 @@ permissions. To understand why an object is not being propagated to a namespace,
...
+[learn how it works](concepts.md#hierarchical-resource-quota) or see an [quickstart example](quickstart.md#hrq)
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:197]: learn -> Learn (capitalized)

File: docs/user-guide/how-to.md [hunk 20]
```diff
@@ -186,6 +187,17 @@ permissions. To understand why an object is not being propagated to a namespace,
...
+
+Note: decimal point values cannot be specified in HRQ (you can't do `cpu: 1.5`). see [#292](https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/292)
```
Comment (adrianludwin) [inline docs/user-guide/how-to.md:199]: see -> See (capitalized)

Maybe also say "You can't say `cpu: 1.5`, but you can say `cpu: 1500m`".

File: docs/user-guide/concepts.md [hunk 21]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+├── organization-b
+│   ├── org-b-team-1
+│   ├── org-b-team-2
+│   ...
+...
+```
+Instead of each team askin from the `cluster-admin` to modify their ResourceQuota, 
+you can make the following namespace hierarchy:
+```
+company-a-restrictions
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:253]: I'd call this `policy` instead of `restrictions`, e.g. `company-a-policy` (here and in the example below)

Also, can you get rid of all the `[s]` in the tree below? These don't have to be subnamespaces (e.g. namespaces with `SubnamespaceAnchor` objects), they just have to be _child_ namespaces.

File: docs/user-guide/concepts.md [hunk 22]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+When you want to give some amount of resources to `team-a`, and want them to be able to
+flexibly use resources in any of their subnamespaces, you create a `HierarchicalResourceQuota`
+in namespace `team-a`. The sum of all resources from all the subnamespaces of the
+members wont be over the amount of resources that is configured in
+`HierarchicalResourceQuota` of namespace `team-a`. All of the reasources of `team-a` are
+equally shared between the applications in their subnamespaces, which is very efficient.
+
+You have the struggle that each time some team calls you and asks you to modify their
+`ResourceQuota` or namespaces structure? HRQ fixes it!
+With the following teams tree:
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:237]: This is a bit too marketing-y, how about this?

```
In addition, you can let an org or team's admin create their own hierarchical
quotas without violating the overall HRQ for their org or team. For example,
if you start with the following structure:
```

File: docs/user-guide/concepts.md [hunk 23]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+│   ├── org-a-team-2
+│   ...
+├── organization-b
+│   ├── org-b-team-1
+│   ├── org-b-team-2
+│   ...
+...
+```
+Instead of each team askin from the `cluster-admin` to modify their ResourceQuota, 
+you can make the following namespace hierarchy:
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:251]: Instead of "you can make the following namespace hierarchy," how about:

```
you can insert an additional "policy" namespace above each level to hold
the policy objects (like hierarchical quota) that the sub-admin _cannot_
change, while giving them permission to create their own quotas in the
lower level namespaces, like this:
```

File: docs/user-guide/concepts.md [hunk 24]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+company-a-restrictions (has HRQ)
+└── [s] company-a
+    ├── [s] org-a-restrictions (has HRQ)
+    │   └── [s] organization-a
+    ├── [s] org-b-restrictions (has HRQ)
+    │   └── [s] organization-b
+    ...
+```
+This way each individual can fairly and securely distribute their resources across
+their members.
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:269]: Add: `Lower-level quotas cannot override more restrictive quotas from ancestor namespaces; the most restrictive quota always wins.`

File: docs/user-guide/concepts.md [hunk 25]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+
+when creating `HierarchicalResourceQuota`, HNC automatically creates `ResourceQuota`
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md]: "when creating" -> "When you create a"

File: docs/user-guide/concepts.md [hunk 26]
```diff
@@ -218,6 +221,61 @@ probably not a great idea.
...
+with name `local-impl-hrq` and propogates it to all subnamespaces. It monitors
+the `ResourceQuotas` in the namespace tree, and sums up their usage. If there are multiple 
+HRQs in the tree, the local `ResourceQuota` will have the combined results of all HRQs
+from the parent namespaces.
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md]: "propogates" -> "propagates"

I'd replace everything after "It monitors" with:

```
This is a part of the internal implementation and shouldn't be modified or
inspected. Use the `kubectl hns hrq` command to inspect hierarchical quotas,
or look at the `HierarchicalResourceQuota` object in the ancestor
namespaces.
```

**Review by adrianludwin — COMMENTED** [review]
Looking good! Can you please squash all your commits into one?

File: docs/user-guide/concepts.md (hunk 3 above)
Comment (zfrhv) [reply docs/user-guide/concepts.md:91]: well i guess i will change it for not only my text then

**Comment by zfrhv** [comment]
i knew i would have so mistakes and typos
but i didnt thought i would be so bad :I 
sorry and thank you

**Comment by adrianludwin** [comment]
> i knew i would have so mistakes and typos but i didnt thought i would be so bad :I sorry and thank you

Not at all, I really appreciate all the work you put into this and all the work I didn't have to do as a result! My only other language is French and I _guarantee_ you I can't write anything this good in French 😁

File: docs/user-guide/concepts.md [hunk 27]
```diff
@@ -218,6 +221,64 @@ probably not a great idea.
 You can create a subnamespace from the command line via `kubectl hns create child
 -n parent`.
 
+<a name="hrq">
+
+### Hierarchical resource quotas (HRQs)
+
```
Comment (adrianludwin) [inline docs/user-guide/concepts.md:227]: Can you please add a note here that HRQ is beta in HNC v1.1? Like this:

https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/docs/user-guide/concepts.md#managed-labels-and-annotations

Similarly in the other two docs?

**Review by adrianludwin — COMMENTED** [review]
lgtm, just one last change and then I'll approve this. Thanks!

**Comment by zfrhv** [comment]
@adrianludwin 
done!
but i didnt checked `test\e2e\quickstart_test.go`, i just added few test
is there a pipeline that runs it? or do i need to install WSL and check it myself?

**Comment by zfrhv** [comment]
welp, i installed go (for some reason i thought its available only on linux), and idk how to run the test files
sorry i tried both `go test quickstart_test.go` and `go run quickstart_test.go`

**Comment by adrianludwin** [comment]
You need to have a cluster with HNC installed, and your kubectl configured to point to it, before you try running the tests. Then you can say (see [here](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/168dd7ebae3a6fd47bdbb649b44d5f96aa1a05c5/Makefile#L339)):

```
HNC_FOCUS=Quickstart make test-e2e # Runs all tests in the "Quickstart" Describe block
```

If that doesn't work, no worries, lmk and I'll try running it for you.
```

**Comment by zfrhv** [comment]
sorry my clusters are airgapped, it will take me few days to test it

**Comment by adrianludwin** [comment]
I just tried this out myself and found the following issues:

* The yaml files cannot have any leading whitespace (see the other embedded yaml files)
  * The yaml files _must_ be indented with two spaces, not tabs (not sure if you had tabs, or if my editor did that when I was fixing the formatting)
* While you're limiting CPU and RAM, you're actually creating `Services`, which don't use any CPU or RAM - services are just load balancers to deployments. So you should try creating a Deployment instead - this is where you'll actually specify how much RAM and CPU you need.

I'd suggest creating the first Deployment in `team-a` and the second in `team-b`

**Comment by zfrhv** [comment]
> The yaml files cannot have any leading whitespace (see the other embedded yaml files)
The yaml files must be indented with two spaces, not tabs (not sure if you had tabs, or if my editor did that when I was fixing the formatting)

yes sorry about that, i did copy pase of the yaml from `quickstart.md` and vscode just replaces my spaces with tabs

> While you're limiting CPU and RAM, you're actually creating Services, which don't use any CPU or RAM - services are just load balancers to deployments. So you should try creating a Deployment instead - this is where you'll actually specify how much RAM and CPU you need.

yes, but the `cpu` and `memory` were just examples, the limitation was with amount of services.
i can create a deployment but i wanted to keep the example as simple as possible, and creating services is a lot easier than deployments.

i removed the cpu and memory from hrq now, you can check again
but if you think that demostrating with actual cpu/memory then i can make a deployment instead of service

**Comment by zfrhv** [comment]
oh no :0
i accidently squashed too many commits

**Comment by zfrhv** [comment]
well that was scary, all good now
anyways, if you want deployments instead of services just let me know

**Comment by adrianludwin** [comment]
It might be even easier to create Secrets instead of Services or Deployments? But I'll try your new changes as-is.

File: docs/user-guide/concepts.md (hunk 7 above)
Comment (mochizuki875) [reply docs/user-guide/concepts.md:233]: @zfrhv 
Is this fixed?(seems to remain "wont")

File: docs/user-guide/concepts.md [hunk 28]
```diff
@@ -218,6 +221,68 @@ probably not a great idea.
...
+This way each individual can fairly and securely distribute their resources across
+their members.
+
+To implement hierarchical quotas, HNC automatically creates `ResourceQuota` objects in each
+affected namespace. This is a part of the internal implementation and shouldn't be modified or
+inspected. Use the `kubectl hns hrq` command to inspect hierarchical quotas,
+or look at the `HierarchicalResourceQuota` object in the ancestor
+namespaces.
+
+Note: Decimal point values cannot be specified in HRQ (you can't do `cpu: 1.5` but you can do `cpu: "1.5"` or `cpu: 1500m`). See [#292](https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/292)
```
Comment (mochizuki875) [inline docs/user-guide/concepts.md:284]: @zfrhv 
thanks for adding it!

File: docs/user-guide/quickstart.md [hunk 29]
```diff
@@ -458,6 +459,98 @@ kubectl delete svc s2 -n service-2
...
+
+Let's assume you own _acme-org_ from the previous example:
+```
+acme-org
+├── [s] team-a
+└── [s] team-b
+```
+
+You can create a `HierarchicalResourceQuota` in namespace `acme-org`, and the sum of 
+all subnamespaces resource usage cant go over what is configured in the hrq.
```
Comment (mochizuki875) [inline docs/user-guide/quickstart.md:478]: @zfrhv 
cant -> can't ?

**Comment by zfrhv** [comment]
> It might be even easier to create Secrets instead of Services or Deployments? But I'll try your new changes as-is.

yes creating secrets/configmaps was my first idea, but kubernetes has few default cm/secrets that are being automatically creates when the namespace is created,
and the amount of configmaps by default in openshift is different than in kubernetes (same with secrets)
so i cant relay on how many secrets or configmaps already exist, unless i start counting them

**Comment by zfrhv** [comment]
i simplified the quickstart a little,
let me know what you think and i will squash the commits

**Comment by adrianludwin** [comment]
Ok, I tried it and it worked with the following changes. Nice work!

```
diff --git a/test/e2e/quickstart_test.go b/test/e2e/quickstart_test.go
index 72cf25d8..7f6f9565 100644
--- a/test/e2e/quickstart_test.go
+++ b/test/e2e/quickstart_test.go
@@ -217,13 +217,13 @@ spec:
                MustApplyYAML(hrq)
 
                // create service in team-a
-               MustRun("kubectl create service clusterip", nsTeamA + "-svc", `--clusterip="None"`, "-n", nsTeamA)
+               MustRun("kubectl create service clusterip", nsTeamA + "-svc", `--clusterip=None`, "-n", nsTeamA)
 
                // show that you can't use resources more than the hrq
-               MustNotRun("kubectl create service clusterip", nsTeamB + "-svc", `--clusterip="None"`, "-n", nsTeamB)
+               MustNotRun("kubectl create service clusterip", nsTeamB + "-svc", `--clusterip=None`, "-n", nsTeamB)
 
                // show hrq usage
-               RunShouldContain("services 1/1", defTimeout, "kubectl hns hrq", "-n", nsOrg)
+               RunShouldContain("services: 1/1", defTimeout, "kubectl hns hrq", "-n", nsOrg)
 
                MustRun("kubectl delete hrq", nsOrg + "-hrq", "-n", nsOrg)
```

Note the changes: no double-quotes around the "None" in `--clusterip=None` (the double-quotes are removed by `bash` when you type it by hand, but there's no shell in this test) and an added colon in `services: 1/1`.

Please go ahead and squash the commits. I'm going to have to enable HRQ by default before these tests will pass though so I'll work on that next.

**Comment by zfrhv** [comment]
done
should i create a new PR for #295 ? (so this doc wouldnt have to wait until that feature would be merge)

**Comment by adrianludwin** [comment]
Actually - let's leave HRQ off in the default version of v1.1, I'll turn it on along with all the tests next.

Can you please disable this test by default (add a "P" before the `It` function so it says `PIt` - "P" for "Pending") and I'll enable it on the master branch next. Then I'll merge. Thanks!

**Comment by adrianludwin** [comment]
Oh, and please add a mention that HRQ is off by default, and you need to use the `hrq.yaml` manifest to enable it? Thanks!

**Comment by zfrhv** [comment]
sure thing!
i added it after the beta warning on each page, hope you like it
(and for some reason the new commit is not added to this PR, even tho i can clearly see it in my repo)

**Comment by rjbez17** [comment]
/ok-to-test
/lgtm

@adrianludwin can approve when ready

**Comment by adrianludwin** [comment]
Thank you for all these changes! Can you please cherry-pick this commit to v1.1 as well? Thanks! :)

/lgtm
/approve
